src/plugins/auto_batch/src/plugin.cpp

```diff
@@ -49,311 +41,320 @@ DeviceInformation Plugin::ParseBatchDevice(const std::string& deviceWithBatch) {
 
         if (batch <= 0) {
-            IE_THROW() << "Batch value for '" << deviceName << "' must be > 0, while " << batch << "is passed";
+            OPENVINO_THROW("Batch value for '", deviceName, "' must be > 0, while ", batch, "is passed");
         }
     }
-    return {deviceName, {{}}, batch};
+    return {deviceName, {{}}, static_cast<uint32_t>(batch)};
 }
 
-DeviceInformation Plugin::ParseMetaDevice(const std::string& devicesBatchCfg,
-                                          const std::map<std::string, std::string>& user_config) const {
-    auto metaDevice = ParseBatchDevice(devicesBatchCfg);
-    metaDevice.config = GetCore()->GetSupportedConfig(metaDevice.device_name, user_config);
-
+DeviceInformation Plugin::parse_meta_device(const std::string& devices_batch_config,
+                                            const ov::AnyMap& user_config) const {
+    auto meta_device = parse_batch_device(devices_batch_config);
+    meta_device.device_config = get_core()->get_supported_property(meta_device.device_name, user_config);
     // check that no irrelevant config-keys left
     for (const auto& k : user_config) {
         const auto& name = k.first;
-        if (metaDevice.config.find(name) == metaDevice.config.end() &&
+        if (meta_device.device_config.find(name) == meta_device.device_config.end() &&
             !ov::util::contains(supported_configKeys, name)) {
-            IE_THROW() << "Unsupported config key: " << name;
+            OPENVINO_THROW("Unsupported config key: ", name);
         }
     }
-    return metaDevice;
+    return meta_device;
 }
 
-InferenceEngine::RemoteContext::Ptr Plugin::CreateContext(const InferenceEngine::ParamMap& remote_properties) {
-    auto cfg = remote_properties;
-    auto it = cfg.find(CONFIG_KEY(AUTO_BATCH_DEVICE_CONFIG));
-    if (it == cfg.end())
-        it = cfg.find(ov::device::priorities.name());
-    if (it == cfg.end())
-        IE_THROW() << "Value for KEY_AUTO_BATCH_DEVICE_CONFIG is not set";
-
-    auto val = it->second.as<std::string>();
-    auto core = GetCore();
-    if (!core)
-        return nullptr;
-    auto metaDevice = ParseMetaDevice(val, std::map<std::string, std::string>());
-    cfg.erase(it);
-    return core->CreateContext(metaDevice.device_name, cfg);
+std::shared_ptr<ov::IRemoteContext> Plugin::create_context(const ov::AnyMap& remote_properties) const {
+    OPENVINO_NOT_IMPLEMENTED;
+    // auto cfg = remote_properties;
+    // auto it = cfg.find(CONFIG_KEY(AUTO_BATCH_DEVICE_CONFIG));
+    // if (it == cfg.end())
+    //     it = cfg.find(ov::device::priorities.name());
+    // if (it == cfg.end())
+    //     IE_THROW() << "Value for KEY_AUTO_BATCH_DEVICE_CONFIG is not set";
+
+    // auto val = it->second.as<std::string>();
+    // auto core = get_core();
+    // if (!core)
+    //     return nullptr;
+    // auto metaDevice = parse_meta_device(val, ov::AnyMap());
+    // cfg.erase(it);
+    // return core->create_context(metaDevice.device_name, cfg);
```

Comment:
@ilya-lavrenov @ilyachur @riverlijunjie @songbell 
Here is the implementation of `create_context()`, the return type is `ov::RemoteContext` from call `core->create_context(metaDevice.device_name, cfg);` but `ov::**I**RemoteContext` is needed. Do we have a way to get the plugin by device name & than call plugin level `create_context()`.

Reply:
Do you mean it's better to have
```c++
ov::IRemoteContext create_context(const std::string& device_name, const AnyMap& args) const override;
```
instead of
```c++
ov::RemoteContext create_context(const std::string& device_name, const AnyMap& args) const override;
```
?

Reply:
Yes

Reply:
used friendly class